backend/modules/parsers/audioparser.py

```diff
@@ -0,0 +1,125 @@
+import json
+from typing import Any, Dict, List
+
+import aiofiles
+import aiohttp
+from langchain.docstore.document import Document
+
+from backend.logger import logger
+from backend.modules.model_gateway.model_gateway import model_gateway
+from backend.modules.parsers.parser import BaseParser
+from backend.modules.parsers.unstructured_io import UnstructuredIoParser
+from backend.types import ModelConfig
+
+
+class AudioParser(BaseParser):
+    """
+    AsyncAudioParser is a parser class for extracting text from audio input.
+
+    {
+        ".mp3": {
+            "name": "AudioParser",
+            "parameters": {
+                "model_configuration": {
+                    "name" : "faster-whisper/Systran/faster-distil-whisper-large-v3"
+                },
+                "max_chunk_size": 2000
+            }
+        }
+    }
+
+    """
+
+    supported_file_extensions = [
+        ".flac",
+        ".mp3",
+        ".mp4",
+        ".mpeg",
+        ".mpga",
+        ".m4a",
+        ".ogg",
+        ".wav",
+        ".webm",
+    ]
+
+    def __init__(
+        self, *, model_configuration: ModelConfig, max_chunk_size: int = 2000, **kwargs
+    ):
+        """
+        Initializes the AsyncAudioParser object.
+        """
+        self.model_configuration = ModelConfig.model_validate(model_configuration)
+        self.audio_processing_svc = None
+        self.max_chunk_size = max_chunk_size
+        super().__init__(**kwargs)
+
+    async def initialize(self):
+        """
+        Initialize the audio processing service.
+        """
+        self.audio_processing_svc = (
+            await model_gateway.get_audio_model_from_model_config(
+                model_name=self.model_configuration.name
+            )
+        )
```

Comment:
This does not really have to be async if we keep get_audio_model_from_model_config sync, then can be part of __init__ itself

Reply:
yaa, now that it is async, I cant use it in init. Will refactor this later.